tests/unit/test_sql_sensor.py

```diff
@@ -0,0 +1,155 @@
+import unittest
+from unittest import mock
+
+from airflow.exceptions import AirflowException
+from airflow_clickhouse_plugin import ClickHouseSqlSensor
+
+
+class ClickHouseSqlSensorTestCase(unittest.TestCase):
+    @mock.patch(
+        'airflow_clickhouse_plugin.sensors.clickhouse_sql_sensor.ClickHouseHook'
+    )
+    def test_clickhouse_sql_sensor_poke(self, mock_hook):
+        op = ClickHouseSqlSensor(task_id='sql_sensor_check', sql='SELECT 1',)
+
+        mock_get_records = mock_hook().get_records
+
+        mock_get_records.return_value = []
+        self.assertFalse(op.poke(None))
+
+        mock_get_records.return_value = [[None]]
+        self.assertFalse(op.poke(None))
+
+        mock_get_records.return_value = [['None']]
+        self.assertTrue(op.poke(None))
+
+        mock_get_records.return_value = [[0.0]]
+        self.assertFalse(op.poke(None))
+
+        mock_get_records.return_value = [[0]]
+        self.assertFalse(op.poke(None))
+
+        mock_get_records.return_value = [['0']]
+        self.assertTrue(op.poke(None))
+
+        mock_get_records.return_value = [['1']]
+        self.assertTrue(op.poke(None))
+
+    @mock.patch(
+        'airflow_clickhouse_plugin.sensors.clickhouse_sql_sensor.ClickHouseHook'
+    )
```

Comment:
I would insist you define a `setUp` or even `setUpClass` method instead of copy-pasting decorator. Like this:
```python
class ClickHouseSqlSensorTestCase(unittest.TestCase):
    _hook_mock: mock.MagicMock
    _hook_patch: mock._patch
    def setUp(self):
        self._hook_patch = mock.patch('airflow_clickhouse_plugin.sensors.clickhouse_sql_sensor.ClickHouseHook')
        self._hook_mock = self._hook_patch.__enter__()
    def tearDown(self):
        self._hook_patch__exit___(None, None, None)
```

This will create a new patched hook on every test. Just like your decorator but without any code duplication.

A more complex example: https://github.com/whisklabs/airflow-clickhouse-plugin/blob/a88b4249a0db0b06866d5a4bedc2e08254773af1/tests/unit/test_hook.py#L74-L102